gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcClientCalls.java

```diff
@@ -40,19 +42,31 @@
  * <p>Package-private for internal use.
  */
 class GrpcClientCalls {
-
   private GrpcClientCalls() {};
 
   public static <RequestT, ResponseT> ClientCall<RequestT, ResponseT> newCall(
-      MethodDescriptor<RequestT, ResponseT> descriptor, ApiCallContext context) {
+      MethodDescriptor<RequestT, ResponseT> descriptor,
+      ApiCallContext context,
+      RequestT request,
+      RequestParamsEncoder<RequestT> paramsEncoder) {
     if (!(context instanceof GrpcCallContext)) {
       throw new IllegalArgumentException(
           "context must be an instance of GrpcCallContext, but found "
               + context.getClass().getName());
     }
+
     GrpcCallContext grpcContext = (GrpcCallContext) context;
     Preconditions.checkNotNull(grpcContext.getChannel());
-    Preconditions.checkNotNull(grpcContext.getCallOptions());
-    return grpcContext.getChannel().newCall(descriptor, grpcContext.getCallOptions());
+
+    CallOptions callOptions = grpcContext.getCallOptions();
+    Preconditions.checkNotNull(callOptions);
+
+    if (paramsEncoder != null) {
+      callOptions =
+          CallOptionsUtil.putRequestParamsDynamicHeaderOption(
+              callOptions, paramsEncoder.encode(request));
+    }
```

Comment:
This doesn't sit right with me for some reason

Reply:
What exactly? GrpcClientCalls with a single static newCall() method in general does not sit right with me (why isn't if, for example an absrtact class extended by streaming direct callables?